src/OptimalControlUtilities/include/BipedalLocomotionControllers/OptimalControlUtilities/ControlProblemElements.h

```diff
@@ -0,0 +1,745 @@
+     * a pair containing the name of the frame in the variableHandler and in the model.
+     * @throw std::runtime_error if the frame is not defined
+     */
+    CentroidalAngularMomentumElement(
+        iDynTree::KinDynComputations& kinDyn,
+        const VariableHandler& handler,
+        const std::vector<std::pair<std::string, std::string>>& framesInContact);
+
+    /**
+     * Set the desired centroidal angular momentum
+     * @param centroidalAngularMomentumVelocity desired angular momentum rate of change
+     * @param centroidalAngularMomentum desired angular momentum
+     */
+    void
+    setDesiredCentroidalAngularMomentum(const iDynTree::Vector3& centroidalAngularMomentumVelocity,
+                                        const iDynTree::Vector3& centroidalAngularMomentum);
+
+    /**
+     * Set the controller gain
+     * @param kp controller gain
+     */
+    void setGain(const double& kp);
+
+    /**
+     * Get (and compute) the element matrix
+     * @return the element matrix
+     */
+    virtual const iDynTree::MatrixDynSize& getA() final;
+
+    /**
+     * Get (and compute) the element vector
+     * @return the element vector
+     */
+    virtual const iDynTree::VectorDynSize& getB() final;
+};
+
+/**
+ * RegularizationElement describes an element used to regularize unknown variables
+ * It is in general used as CostFunctionElement of the form  \f$ x^\top A^\top A x \f$
+ */
+class RegularizationElement : public CostFunctionOrEqualityConstraintElement
+{
+public:
+    /**
+     * Constructor.
+     * @param kinDyn an iDynTree kinDyn computation object
+     * @param handler the variable handler object
+     * @param variableName name of the unknown variable that should minimized
+     * @throw std::runtime_error if the variableName is not defined in the handler
+     */
+    RegularizationElement(iDynTree::KinDynComputations& kinDyn,
+                          const VariableHandler& handler,
+                          const std::string& variableName);
+
+    /**
+     * Get (and compute) the element matrix
+     * @return the element matrix
+     */
+    virtual const iDynTree::MatrixDynSize& getA() final;
+
+    /**
+     * Get (and compute) the element vector
+     * @return the element vector
+     */
+    virtual const iDynTree::VectorDynSize& getB() override;
+};
+
+/**
+ * RegularizationWithControlElement describes an element used to regularize unknown variables using
+ * a control law It is in general used as CostFunctionElement of the form  \f$ x^\top A^\top A x +
+ * g^\top x \f$
+ */
+class RegularizationWithControlElement : public RegularizationElement
+{
+    iDynTree::VectorDynSize m_kp; /**< Proportional gain */
+    iDynTree::VectorDynSize m_kd; /**< Derivative gain */
+
+    iDynTree::VectorDynSize m_desiredPosition; /**< Desired position. */
+    iDynTree::VectorDynSize m_desiredVelocity; /**< Desired velocity. */
+    iDynTree::VectorDynSize m_desiredAcceleration; /**< Desired acceleration. */
```

Comment:
Why not using a PID?

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/3/commits/0484ff383119bbbd3155db8bc3e6bbc6d71ce5c1